core/src/oauth/providers/zendesk.rs

```diff
@@ -45,8 +48,20 @@ impl Provider for ZendeskConnectionProvider {
             "scope": "read"
         });
 
+        let sub_domain = match connection.metadata()["extra_config"].as_str() {
+            Some(c) => {
+                if !ZENDESK_SUBDOMAIN_RE.is_match(c) {
+                    Err(anyhow!("Zendesk extra_config subdomain format invalid"))?
+                }
+                c
+            }
+            None => Err(anyhow!("Zendesk extra_config subdomain is missing"))?,
+        };
+
+        let url = format!("https://{}.zendesk.com/oauth/tokens", sub_domain);
+
         let req = reqwest::Client::new()
-            .post("https://d3v-dust.zendesk.com/oauth/tokens")
+            .post(url)
```

Comment:
Any chance we can validate the subdomain in the response as well?

Reply:
wdym in the response? we're hitting it, there nothing to validate no?

Reply:
Was mainly wondering if we need to add some specific checks in the response handling.

Reply:
I don't believe we need to add anything here.